game/glshader.cc

```diff
@@ -29,7 +29,7 @@ void Shader::dumpInfoLog(GLuint id) {
 	if (glIsShader(id)) glGetShaderiv(id, GL_INFO_LOG_LENGTH, &maxLength);
 	else glGetProgramiv(id, GL_INFO_LOG_LENGTH, &maxLength);
 
-	char infoLog[maxLength];
+	char* infoLog = new char[maxLength];
```

Comment:
Do not (ever) use new/delete[]. I suppose that std::vector should do fine here, if the buffer is too large for stack.

Reply:
Putting the string on the stack would be fine, but the syntax is invalid for standard C++ (I'm aware that GCC allows this, but MSVC doesn't).

Reply:
What about a unique_ptr?

Reply:
It is unfortunate that VLAs (supported since C99 and planned for C++14) didn't make it into standard, and AFAIK there are no alternatives either. std::vector is the normal (least confusing) solution here. Unique ptrs only got array support in C++17 (do we still support C++14?), and it is hard to see any benefit over vectors.